localstack/aws/protocol/serializer.py

```diff
@@ -655,16 +654,17 @@ def _serialize_header_value(self, shape: Shape, value: any):
         """Serializes a value for the location trait "header"."""
         if shape.type_name == "timestamp":
             datetime_obj = parse_to_aware_datetime(value)
-            timestamp = calendar.timegm(datetime_obj.utctimetuple())
             timestamp_format = shape.serialization.get(
                 "timestampFormat", self.HEADER_TIMESTAMP_FORMAT
             )
-            return self._convert_timestamp_to_str(timestamp, timestamp_format)
+            return self._convert_timestamp_to_str(datetime_obj, timestamp_format)
         elif shape.type_name == "list":
             converted_value = [
                 self._serialize_header_value(shape.member, v) for v in value if v is not None
             ]
             return ",".join(converted_value)
+        elif shape.type_name == "boolean":
+            return "true" if value else "false"
```

Comment:
are we testing this somewhere? it seems the test that was added does not have a boolean flag. moreover, are you sure the check is correct? what values can `value` have?

Reply:
Sorry, this change should have been separated.
It came up with the test which has been added (or actually where the skip marker has been removed).
It basically sets a boolean header value to the string "true" if the given value for the shape is "truthy".
I clarified that with another comment and added a specific section in the restxml header test.